esp-hal-common/src/gpio.rs

```diff
@@ -1909,18 +1909,41 @@ mod asynch {
 
     #[interrupt]
     unsafe fn GPIO() {
-        // TODO how to handle dual core reg access
-        // we need to check which core the interrupt is currently firing on
-        // and only fire interrupts registered for that core
-        type Bank0 = SingleCoreInteruptStatusRegisterAccessBank0;
-        #[cfg(any(esp32, esp32s2, esp32s3))]
-        type Bank1 = SingleCoreInteruptStatusRegisterAccessBank1;
-        let mut intrs = Bank0::pro_cpu_interrupt_status_read() as u64;
+        // Whilst the S3 is a dual core chip, it shares the enable registers between
+        // cores so treat it as a single core device
+        cfg_if::cfg_if! {
+            if #[cfg(any(esp32c2, esp32c3, esp32c6, esp32h2, esp32s2, esp32s3))] {
```

Comment:
Sorry for nitpicking, wouldn't be something like 

```suggestion
            if #[cfg(not(esp32))] {
```

more readable?